common-app-lib/src/cmr/common_app/services/search/elastic_search_index.clj

```diff
@@ -48,39 +49,101 @@
   [concept-type fields]
   (map #(q2e/query-field->elastic-field (keyword %) concept-type) fields))
 
+(defn- query->execution-params
+  "Returns the Elasticsearch execution parameters extracted from the query. These are the
+  actual ES parameters as epxected by the elastisch library. The :scroll-id parameter is special
+  and is stripped out before calling elastisch to determine whether a normal search call or a
+  scroll call should be made."
+  [query]
+  (let [{:keys [page-size offset concept-type aggregations highlights scroll scroll-id]} query
+        scroll-timeout (when scroll (es-config/elastic-scroll-timeout))
+        search-type (if scroll
+                        (es-config/elastic-scroll-search-type)
+                        "query_then_fetch")
+        sort-params (q2e/query->sort-params query)
+        fields (query-fields->elastic-fields
```

Comment:
Does `query-fields->elastic-fields` do any sort of checks to avert injection attacks?